nomad/parts/plugins/x-nomad.py

```diff
@@ -0,0 +1,49 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2015, 2016 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""This is a modifyied version of snapcraft's go and make plugins.
+
+It uses code from  the go plugin to pull, and the make plugin to build and
+install.
+
+"""
+import os
+import shutil
+
+from snapcraft import BasePlugin
+from snapcraft.plugins import go
+
+
+class NomadPlugin(go.GoPlugin):
+
+    def build(self):
+        BasePlugin.build(self)
```

Comment:
First of all, shouldn't that be `super(go.GoPlugin, self).build()` :stuck_out_tongue: ?

Yikes, so this thing requires golang and a go workspace, but then builds with make? First of all, I don't really have a problem with the inheritance skip like that if it reduces duplication. But does nomad need something really special from the go plugin (e.g. the importpath support, or go-packages)? Or really just the setup of the workspace? If the latter, I say scratch the inheritance and write your own pull step. You can better guarantee its functionality, and make it work specifically for nomad.